### DISCUSSION TITLE
Download store data to a file (export and import)

### DISCUSSION THREAD

**Discussion by lsdebianchi** [tinyplex/tinybase#164] [Q&A] [answered]
Hello.

Is there a way to download a `store` as a file while on the browser?
And in reverse, is there a way to load/import the `store` data from a file (always in the context of a browser)?

If not, I think that a utility function (or a persister type?) that allows to do that would be a great addition, as this library is already very local-first.

People may easily create their own export/import implementations, but I am sure that a build-in would help DX and would have other synergies with the librar's features (persister? sync? formats?).

In my case I would need this for a fully offline app (a personal tracker/logger) that, once loaded from the web, works fully offline. So all the data is persisted with TinyBase+IndexDb. I felt the need to allow users to export/import for ownership/migration/backup reason.

Btw, nice library ⭐ well done!

**Comment by jamesgpearce** [comment, marked as answer]
Do the getJson and setJson methods work for this maybe?

**Reply by lsdebianchi** [reply]
Ah yes! My bad, I couldn't find it the in the docs... I should have been more curious exploring the code itself.

`getJson`and `setJson` are perfect.

Thanks for the quick answer!
